### PR TITLE
Add garbage collector - Prep code

### PR DESCRIPTION
**Pull request by mokagio**
This PR adds all the noisy infrastructure code on which the garbage collector script will rely, so that it'll be easier to review its code. It's purpose is to make #6 easier to review.

- Bundler setup
- RSpec setup
- JSON fixtures

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+
+ruby File.read('.ruby-version').strip
```
Comment (AliSoftware) [inline Gemfile:5]: smart!

File: spec/github_client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+def get_pr_response(state)
+  # This is only a fragment of the full API response with the data relevant the
+  # job we have to do.
```
Comment (AliSoftware) [inline spec/github_client_spec.rb]:
```suggestion
  # This is only a fragment of the full API response with the data relevant for
  # the job we have to do.
```

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+=begin
+  # This allows you to limit a spec run to individual examples or groups
+  # you care about by tagging them with `:focus` metadata. When nothing
+  # is tagged with `:focus`, all examples get run. RSpec also provides
+  # aliases for `it`, `describe`, and `context` that include `:focus`
+  # metadata: `fit`, `fdescribe` and `fcontext`, respectively.
+  config.filter_run_when_matching :focus
```
Comment (AliSoftware) [inline spec/spec_helper.rb:54]: TIL! worth enabling imho

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  # Limits the available syntax to the non-monkey patched syntax that is
+  # recommended. For more details, see:
+  #   - http://rspec.info/blog/2012/06/rspecs-new-expectation-syntax/
+  #   - http://www.teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/
+  #   - http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#zero-monkey-patching-mode
+  config.disable_monkey_patching!
+
+  # This setting enables warnings. It's recommended, but in some cases may
+  # be too noisy due to issues in dependencies.
+  config.warnings = true
```
Comment (AliSoftware) [inline spec/spec_helper.rb:70]: worth enabling imho

**Review by AliSoftware — APPROVED** [review]
LGTM, although I'd probably enable some of the optional/commented config options for RSpec here for good measure (better use the good practices from the start!)

File: spec/spec_helper.rb (hunk 4 above)
Comment (loremattei) [reply spec/spec_helper.rb:70]: I agree. Especially at this early stage.

**Review by loremattei — APPROVED** [review]
Other than Olivier's suggestion, this looks good to me.

**Comment by mokagio** [comment]
Thanks folks for the reviews.

I'll admit that, in the midsts of trying to get the tests in place, I glossed over the RSpec recommendations. Definitely work enabling, so thank you very much for pointing them out 🙇

**Pull request merged**